### PR TITLE
RavenDB-18078 Highlight - negative for start fregment index, better h…

### PR DESCRIPTION
**Pull request by ayende**
…ighlight support for map/reduce indexes

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18078 

### Additional description

Handle highlight terms that start with `, ` 

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Ensured. Please explain how has it been implemented?

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: test/SlowTests/Tests/Issues/RavenDB-18078.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            {
+                var query = session.Advanced.AsyncDocumentQuery<Index_TESTIndexOff.Result, Index_TESTIndexOff>()
+                    .WaitForNonStaleResults()
+                    .Highlight("Highlight", 18, 1, out Highlightings titleHighlighting)
+                    .Search("Highlight", "aa");
+                    
+
+             
+                _ = await query
+                    .SingleAsync();
```
Comment (arekpalinski) [inline test/SlowTests/Tests/Issues/RavenDB-18078.cs:127]: No need to assert anything?

Comment (ayende) [reply test/SlowTests/Tests/Issues/RavenDB-18078.cs:127]: No, this would throw an error previously

Comment (arekpalinski) [reply test/SlowTests/Tests/Issues/RavenDB-18078.cs:127]: Yes, but don't we care about the correctness of the returned highlights?

Comment (ayende) [reply test/SlowTests/Tests/Issues/RavenDB-18078.cs:127]: No, that is tested elsewhere

Comment (arekpalinski) [reply test/SlowTests/Tests/Issues/RavenDB-18078.cs:127]: ok

**Review by ppekrol — APPROVED** [review]

**Review by haludi — APPROVED** [review]

**Pull request merged**